waku/v2/node/wakuoptions.go

```diff
@@ -373,12 +372,15 @@ func WithPeerStoreCapacity(capacity int) WakuNodeOption {
 }
 
 // WithDiscoveryV5 is a WakuOption used to enable DiscV5 peer discovery
-func WithDiscoveryV5(udpPort uint, bootnodes []*enode.Node, autoUpdate bool) WakuNodeOption {
+func WithDiscoveryV5(discv5Params discv5.DiscV5Parameters) WakuNodeOption {
 	return func(params *WakuNodeParameters) error {
-		params.enableDiscV5 = true
-		params.udpPort = udpPort
-		params.discV5bootnodes = bootnodes
-		params.discV5autoUpdate = autoUpdate
+		params.discv5Params = discv5Params
+		return nil
+	}
+}
+func WithDiscoveredNodes(discoveredNodes []dnsdisc.DiscoveredNode) WakuNodeOption {
```

Comment:
IMO we're unnecesarily coupling DNSDiscovery with Discv5. DiscV5 is only concerned with ENRs for its bootnode setup